interfaces/api/v1/core/settings.py

```
@@ -151,3 +142,30 @@ async def _fetch_openai_models(api_key: str, base_url: str) -> List[str]:
+        url = f"{base}/models"
+    else:
+        url = f"{base}/v1/models"
+    headers = {"x-api-key": api_key, "anthropic-version": "2023-06-01"}
+    try:
+        async with httpx.AsyncClient(timeout=10) as client:
+            resp = await client.get(url, headers=headers, params={"limit": 1000})
+            resp.raise_for_status()
+            data = resp.json()
+            models = data.get("data", [])
+            return sorted(m["id"] for m in models if "id" in m)
+    except Exception as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)